packages/dashboard/src/app/views/myStories/content/storiesView/index.js

```diff
@@ -174,12 +176,29 @@ function StoriesView({
       menuItemActions: {
         default: () => setContextMenuId(-1),
         [STORY_CONTEXT_MENU_ACTIONS.COPY_STORY_LINK]: handleCopyStoryLink,
-        [STORY_CONTEXT_MENU_ACTIONS.DELETE]: handleDeleteStory,
-        [STORY_CONTEXT_MENU_ACTIONS.DUPLICATE]: handleDuplicateStory,
+        ...(apiCallbacks?.trashStory && {
+          [STORY_CONTEXT_MENU_ACTIONS.DELETE]: handleDeleteStory,
+        }),
+        ...(apiCallbacks?.duplicateStory && {
+          [STORY_CONTEXT_MENU_ACTIONS.DUPLICATE]: handleDuplicateStory,
+        }),
         [STORY_CONTEXT_MENU_ACTIONS.OPEN_STORY_LINK]: handleOpenStoryInEditor,
-        [STORY_CONTEXT_MENU_ACTIONS.RENAME]: handleRenameStory,
+        ...(apiCallbacks?.updateStory && {
+          [STORY_CONTEXT_MENU_ACTIONS.RENAME]: handleRenameStory,
+        }),
```

Comment:
Are these checks for the `menuItemActions` really needed? Menu items without an API callback are filtered out below anyway.